examples/common/gconfig/FairVMCConfig.cxx

```diff
@@ -18,18 +18,26 @@
 #include <TVirtualMC.h>
 
 FairVMCConfig::FairVMCConfig()
-    : FairYamlVMCConfig()
+    : FairXmlVMCConfig()
 {}
 
 FairVMCConfig::~FairVMCConfig() {}
 
 void FairVMCConfig::SetupStack()
 {
     FairStack *stack = new FairStack(1000);
-    if (fYamlConfig["FairStack_StoreSecondaries"])
-        stack->StoreSecondaries(fYamlConfig["FairStack_StoreSecondaries"].as<bool>());
-    if (fYamlConfig["FairStack_MinPoints"])
-        stack->SetMinPoints(fYamlConfig["FairStack_MinPoints"].as<int>());
+
+    XMLNodePointer_t xmlNode = 0;
+    if ((xmlNode = GetPointer("SIM:FairStack:StoreSecondaries", fXmlConfig))) {
+        LOG(debug) << "Setting SIM:FairStack:StoreSecondaries to \"" << (ConvertTo<bool>(xmlNode) ? "true" : "false")
+                   << "\"";
+        stack->StoreSecondaries(ConvertTo<bool>(xmlNode));
+    }
+    if ((xmlNode = GetPointer("SIM:FairStack:MinPoints", fXmlConfig))) {
+        LOG(debug) << "Setting SIM:FairStack:MinPoints to \"" << ConvertTo<float>(xmlNode) << "\"";
+        stack->SetMinPoints(ConvertTo<float>(xmlNode));
+    }
+
     TVirtualMC::GetMC()->SetStack(stack);
```

Comment:
```suggestion
    auto stack = std::make_unique<FairStack>(1000);

    if (fXmlConfig) {
        if (auto const storeSecondariesNode = GetPointer("SIM:FairStack:StoreSecondaries", fXmlConfig);
            node)
        {
            auto const storeSecondaries = ConvertTo<bool>(storeSecondariesNode);
            LOG(debug) << "Setting SIM:FairStack:StoreSecondaries to \"" << (storeSecondaries ? "true" : "false")
                       << "\"";
            stack->StoreSecondaries(storeSecondaries);
        }
        if (auto const minPointsNode = GetPointer("SIM:FairStack:MinPoints", fXmlConfig);
            minPointsNode)
        {
            auto const minPoints = ConvertTo<float>(minPointsNode);
            LOG(debug) << "Setting SIM:FairStack:MinPoints to \"" << minPoints << "\"";
            stack->SetMinPoints(minPoints);
        }
    } else {
        // Handle this case, e.g. set defaults? warn, error whatever...
    }

    TVirtualMC::GetMC()->SetStack(stack.release());
```
1. Do not use owning raw pointer! (Specifically, nothing that is called here is `noexcept`, so `stack` would leak if an exception is thrown)
2. Handle the case when `fXmlConfig` is `nullptr`!
3. Call the `ConvertTo` functions only once!
4. Do not repeat the type name in the variable name! Prefer readable variable names.

---

The `FairXmlVMCConfig::GetPointer(..., XMLNodePointer_t, ...)` feels awkwardly low level to be used here. I suggest to add a more high-level API, e.g.
```cpp
template <typename T>
auto FairXmlVMCConfig::GetValue<T>(std::string const & path, bool verbose = false)
{
    if (fXmlConfig) {
        if (auto const node = GetPointer(path, fXmlConfig);
            node)
        {
            return ConvertTo<T>(node, verbose);
        }
    }

    // Handle this case, e.g. error, throw whatever...
}
```

then the above code simplifies to:

```cpp
#include <iomanip>   // for std::quoted
// ...


    auto const storeSecondaries = GetValue<bool>("SIM:FairStack:StoreSecondaries");
    auto const minPoints = GetValue<float>("SIM:FairStack:MinPoints");
    auto stack = std::make_unique<FairStack>(1000);

    LOG(debug) << "Setting SIM:FairStack:StoreSecondaries to " << std::quoted(storeSecondaries ? "true" : "false");
    stack->StoreSecondaries(storeSecondaries);

    LOG(debug) << "Setting SIM:FairStack:MinPoints to " << std::quoted(minPoints);
    stack->SetMinPoints(minPoints);

    TVirtualMC::GetMC()->SetStack(stack.release());
    // or, if we also add an overload `SetStack(std::unique_ptr<FairStack>)`
    TVirtualMC::GetMC()->SetStack(std::move(stack));
```